gn.yaml

```diff
@@ -0,0 +1,39 @@
+package:
+  name: gn
+  version: 0.0_git20231222
+  epoch: 0
+  description: "Meta-build system that generates build files for Ninja"
+  copyright:
+    - license: BSD-3-Clause
+
+environment:
+  contents:
+    packages:
+      - build-base
+      - busybox
+      - ca-certificates-bundle
+      - clang
+      - git
+      # Doesn't work on 3.12
+      - python-3.11
```

Comment:
seems to work with python 3.12 added   `./out/gn_unittests` to see if all the tests are passing 


```
package:
  name: gn
  version: 0_git20230709
  epoch: 0
  description: Meta-build system that generates build files for Ninja
  copyright:
    - license: BSD-3-Clause

environment:
  contents:
    packages:
      - busybox
      - ca-certificates-bundle
      - build-base
      - automake
      - autoconf
      - python3
      - samurai
      - git
      - clang

vars:
  commit: 85944ebc24a90ec1e489e85a46fdc68542c3146f 

pipeline:
  - runs: |
      git clone https://gn.googlesource.com/gn
      cd gn
      git checkout ${{vars.commit}}

  - name: Configure and build
    runs: |
      cd gn
      git status
      python3 build/gen.py
      #python3 ./build/gen.py --no-last-commit-position --no-static-libstdc++ --no-strip --allow-warnings
      ninja -C out
      install -dm755 "${{targets.destdir}}"/usr/bin
      install -m755 out/gn "${{targets.destdir}}"/usr/bin/gn
      ./out/gn_unittests

  - uses: strip
  ```

Reply:
The python3 build/gen.py part fails for me - there's a deprecated regex in it, but maybe the flags you passed to gen.py are different?